models/cohortdata.go

```diff
@@ -116,18 +117,34 @@ func (h CohortData) RetrieveHistogramDataBySourceIdAndCohortIdAndConceptIdsAndCo
 	return cohortData, meta_result.Error
 }
 
-func (h CohortData) RetrieveBarGraphDataBySourceIdAndCohortIdAndConceptIds(sourceId int, cohortDefinitionId int, conceptId int64) ([]*OrdinalGroupData, error) {
+func (h CohortData) RetrieveHistogramDataBySourceIdAndConceptId(sourceId int, histogramConceptId int64) ([]*PersonConceptAndValue, error) {
 	var dataSourceModel = new(Source)
 	omopDataSource := dataSourceModel.GetDataSource(sourceId, Omop)
+	var cohortData []*PersonConceptAndValue
 
 	// get the observations for the subjects and the concepts, to build up the data rows to return:
-	var cohortData []*OrdinalGroupData
+	query := omopDataSource.Db.Table(omopDataSource.Schema+".observation as observation").
+		Select("distinct(observation.person_id), observation.observation_concept_id as concept_id, observation.value_as_number as concept_value_as_number").
+		Where("observation.observation_concept_id = ?", histogramConceptId).
+		Where("observation.value_as_number is not null")
 
-	query := omopDataSource.Db.Table(omopDataSource.Schema+".observation_continuous as observation"+omopDataSource.GetViewDirective()).
+	query, cancel := utils.AddTimeoutToQuery(query)
+	defer cancel()
+	meta_result := query.Scan(&cohortData)
+	return cohortData, meta_result.Error
+}
+
+func (h CohortData) RetrieveBarGraphDataBySourceIdAndCohortIdAndConceptIds(sourceId int, conceptId int64) ([]*NominalGroupData, error) {
```

Comment:
in the interface signature of this method, the 2nd variable is called `histogramConceptId`

Reply:
fixed